scripts/deploy.ts

```diff
@@ -1,25 +1,49 @@
 import { ethers } from "hardhat";
 
 async function main() {
+  // Get the ContractFactory and Signers here.
+  const TestERC20 = await ethers.getContractFactory("TestERC20");
+  const GammaPoolFactory = await ethers.getContractFactory("TestGammaPoolFactory");
+  const PositionManager = await ethers.getContractFactory("PositionManager");
+  const GammaPool = await ethers.getContractFactory("TestGammaPool");
   const [owner, addr1, addr2, addr3, addr4] = await ethers.getSigners();
 
-  const Factory = await ethers.getContractFactory("TestGammaPoolFactory");
-  const factory = await Factory.deploy(owner.address, addr1.address, addr2.address, addr3.address);
+  // To deploy our contract, we just have to call Token.deploy() and await
+  // for it to be deployed(), which happens onces its transaction has been
+  // mined.
+  const tokenA = await TestERC20.deploy("Test Token A", "TOKA");
+  const tokenB = await TestERC20.deploy("Test Token B", "TOKB");
+  const cfmm = await TestERC20.deploy("CFMM LP Token", "LP_CFMM");
+  const WETH = await TestERC20.deploy("WETH", "WETH");
+  const factory = await GammaPoolFactory.deploy(owner.address, addr1.address, addr2.address, addr3.address);
 
-  const Erc20 = await ethers.getContractFactory("TestERC20");
-  const erc20 = await Erc20.deploy("WETH", "WETH");
+  await tokenA.deployed();
+  await tokenB.deployed();
+  await cfmm.deployed();
+  await factory.deployed();
 
-  const GammaPool = await ethers.getContractFactory("TestGammaPool");
   const COMPUTED_INIT_CODE_HASH = ethers.utils.keccak256(
     GammaPool.bytecode
   );
 
-  const PositionManager = await ethers.getContractFactory("PositionManager");
-  const positionManager = await PositionManager.deploy(factory.address, erc20.address, COMPUTED_INIT_CODE_HASH);
-    
+  const positionManager = await PositionManager.deploy(factory.address, WETH.address, COMPUTED_INIT_CODE_HASH);
+
   await positionManager.deployed();
 
   console.log("PositionManager deployed to:", positionManager.address);
+
+  const createPoolParams = {
+    cfmm: cfmm.address,
+    protocol: 1,
+    tokens: [tokenA.address, tokenB.address]
+  };
+
+  const gammaPoolAddr = await factory.createPool(createPoolParams);
```

Comment:
Don't use gammaPoolAddr.to. Instead, get the pool id from the event the following way. You just have to check that the object you are referencing is not null to avoid typescript errors.

const res = await (await factory.createPool(createPoolParams)).wait();
if(res.events) {
  if(res.events[1].args) {
    console.log("pool >> " + res.events[1].args.pool);
  }
}

Reply:
Updated.